File: Backend/BancoAPI/BancoAPI/Application/DTOs/MovimientoDto.cs
using BancoAPI.Domain.Enums;
using System.ComponentModel.DataAnnotations;

namespace BancoAPI.Application.DTOs
{
    public class MovimientoDto
    {
        public long id { get; set; }

        [Required(ErrorMessage = "La fecha es requerida")]
        public DateTime fecha { get; set; }

        [Required(ErrorMessage = "El tipo de movimiento es requerido")]
        public TipoMovimiento tipoMovimiento { get; set; } // "Debito" o "Credito"

        [Required(ErrorMessage = "El valor es requerido")]
        public decimal valor { get; set; }

        public decimal saldo { get; set; }

        [Required(ErrorMessage = "La cuenta es requerida")]
        public CuentaResumida cuenta { get; set; }
        public class CuentaResumida
        {
            public long id { get; set; }

            [Required(ErrorMessage = "El número de cuenta es requerido")]
            [StringLength(50, ErrorMessage = "El número de cuenta no debe exceder los 50 caracteres")]
            public string numeroCuenta { get; set; }
        }
    }
}

File: Frontend/banco-devsu/src/app/core/dtos/movimiento.dto.ts
import { TipoMovimiento } from '../enums/tipo-movimiento.enum';

export class MovimientoDto {
  public id: number = 0;
  public fecha: Date = new Date();
  public tipoMovimiento: TipoMovimiento = TipoMovimiento.Credito;
  public valor: number = 0;
  public saldo: number = 0;
  public cuenta: MovimientoDto.CuentaResumida = new MovimientoDto.CuentaResumida();
}

export namespace MovimientoDto {
  export class CuentaResumida {
    public id: number = 0;
    public numeroCuenta: string = '';
  }
}
